bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ValueComparisonUtils.java

```diff
@@ -99,6 +99,219 @@ public static boolean compareValueGreaterThanOrEqual(Object lhsValue, Object rhs
         }
     }
 
+    /**
```

Comment:
Instead of introducing different methods at runtime, can't we desugar `i < j` to `<int> i < j` and likewise for other types?

I think eventually we would be resolving `int|(int|int|int)` as `int`.